vortex-array/src/compute/take.rs

```diff
@@ -1,15 +1,22 @@
 use log::info;
 use vortex_error::{vortex_bail, vortex_err, VortexResult};
 
+use crate::stats::{ArrayStatistics, Stat};
 use crate::{ArrayDType as _, ArrayData, IntoCanonical as _};
 
+#[derive(Default, Debug, Clone, Copy)]
+pub struct TakeOptions {
+    pub skip_bounds_check: bool,
+}
```

Comment:
being able to toggle off bounds checking within a safe function feels kinda off...

Reply:
I noticed an improvement for a couple of benchmarks, so figured I'd add it. Initially it was local within Bool::take implementation, but seemed a shame not to generalize the logic. 

What we could do, is to make TakeOptions public (in case we add any more), but then make the setter for skip_bounds_check non-public, meaning only the take entry-point function has the ability to set it?

Reply:
I guess the other problem here is it means we have to trust the stats from the file?